### PR TITLE
Add an UNTRUSTED_ONLY compile option to the common directory

### PR DESCRIPTION
**Pull request by byron-marohn**
Use this like:
	cmake -DUNTRUSTED_ONLY=ON ..

This is useful to compile and use the common libraries on a system that doesn't support SGX

Signed-off-by: Byron Marohn <byron.marohn@intel.com>

### CODE REVIEW

**Comment by byron-marohn** [comment]
@cmickeyb This is ready to review/merge. This should be an easy one - it does not change the "standard" build in any way.

File: common/CMakeLists.txt [hunk 1]
```diff
@@ -76,6 +81,8 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
 link_directories(${CMAKE_BINARY_DIR})
 
-ADD_SUBDIRECTORY (packages/tinyscheme)
-ADD_SUBDIRECTORY (interpreter)
+if(NOT UNTRUSTED_ONLY)
+	ADD_SUBDIRECTORY (packages/tinyscheme)
+	ADD_SUBDIRECTORY (interpreter)
+endif()
 ADD_SUBDIRECTORY (tests)
```
Comment (cmickeyb) [inline common/CMakeLists.txt]: you have to build these anyway. we need the tinyscheme shared library for testing.

**Comment by byron-marohn** [comment]
Spent some time looking at this and I don't think the one requested change should be made. 

The tinyscheme package built here (disabled for UNTRUSTED_ONLY) is the "trusted" version used only inside the enclave. It has -nostdinc options and only uses SGX include files. It will not be possible to run on a system without SGX libraries. Same thing for everything in the interpreter directory - everything there uses only SGX include files and has -nostdinc. 

This PR is for a way to build the common directory on a system that does not have SGX installed. A handful of SGX header files are still used by the build (for error codes, etc.) so the SGX_SDK environment variable is still required - but none of the libs are needed. 

The untrusted test does indeed pass without compiling these two components.

**Comment by cmickeyb** [comment]
pcontract.so must be built when untrusted (that's the ONLY time its built). i doubt that any tests break because pcontract.so is really only used for debugging contracts without an enclave.

**Pull request merged**